go/vt/vtgate/planbuilder/testdata/memory_sort_cases.json

```diff
@@ -720,49 +729,58 @@
       "QueryType": "SELECT",
       "Original": "select user.col1 as a, user.col2, music.col3 from user join music on user.id = music.id where user.id = 1 order by 1 asc, 3 desc, 2 asc",
       "Instructions": {
-        "OperatorType": "Sort",
-        "Variant": "Memory",
-        "OrderBy": "(0|3) ASC, (2|4) DESC, (1|5) ASC",
-        "ResultColumns": 3,
+        "OperatorType": "SimpleProjection",
+        "Columns": [
+          0,
+          1,
+          2
+        ],
         "Inputs": [
           {
-            "OperatorType": "Join",
-            "Variant": "Join",
-            "JoinColumnIndexes": "L:1,L:2,R:0,L:3,R:1,L:4",
-            "JoinVars": {
-              "user_id": 0
-            },
-            "TableName": "`user`_music",
+            "OperatorType": "Sort",
+            "Variant": "Memory",
+            "OrderBy": "(3|4) ASC, (2|5) DESC, (1|6) ASC",
```

Comment:
truncation at the end could avoid projection at operation.

Reply:
https://github.com/vitessio/vitess/pull/12954#issuecomment-1520004048